src/eventsource.js

```diff
@@ -92,6 +92,11 @@ $.EventSource.prototype = {
      * @param {Number} [priority=0] - Handler priority. By default, all priorities are 0. Higher number = priority.
      */
     addHandler: function ( eventName, handler, userData, priority ) {
+        let errorMsg = this._validateEvent(eventName);
+        if(errorMsg){
+            $.console.error(`Error adding event handler: ${errorMsg}`);
```

Comment:
I would just print error message right away instead of returning it like this. `else` after `return` is not something I like to see either. Finally, hardcoding event names on a class that has no relation to a certain event (the source can be anything, I use this class for event support on various instances) so I would propose doing some general checker support, e.g.

````
setEventValidation(eventName, predicate) {...}
_validateEvent(eventName){
       const event = _getEventValudation(eventName);
       if (event && !event())  {
             //return or console error
       }
       return null;
}
````

Reply:
I understand, but I have the experience that stuff merged here stays in the library for years :)

Reply:
Yup, I agree. I think the new approach is clearer. Let me know what you think.